src/parallel/boundaries/BoundaryUtils.cpp

```diff
@@ -0,0 +1,296 @@
+BoundaryUtils::convertDimensionToStringAbs(DimensionType dimension) {
+  return convertDimensionToString(dimension).substr(1, 1);
+}
+
+int BoundaryUtils::convertDimensionToNumeric(DimensionType dimension) {
+  switch (dimension) {
+  case DimensionType::POSX:
+    return 1;
+
+  case DimensionType::POSY:
+    return 2;
+
+  case DimensionType::POSZ:
+    return 3;
+
+  case DimensionType::NEGX:
+    return -1;
+
+  case DimensionType::NEGY:
+    return -2;
+
+  case DimensionType::NEGZ:
+    return -3;
+
+  default:
+    Log::global_log->error()
+        << "Invalid dimension passed for enum conversion" << std::endl;
+    mardyn_exit(1);
+    return 0;
+  }
+}
+
+int BoundaryUtils::convertDimensionToNumericAbs(DimensionType dimension) {
+  int toReturn = convertDimensionToNumeric(dimension);
+  return std::abs(toReturn);
+}
+
+int BoundaryUtils::convertDimensionToLS1Dims(DimensionType dimension) {
+  return convertDimensionToNumericAbs(dimension) - 1;
+}
+
+BoundaryUtils::BoundaryType
+BoundaryUtils::convertStringToBoundary(std::string boundary) {
+  if (boundary == "periodic")
+    return BoundaryType::PERIODIC_OR_LOCAL;
+  if (boundary == "reflecting" || boundary == "reflective")
+    return BoundaryType::REFLECTING;
+  if (boundary == "outflow")
+    return BoundaryType::OUTFLOW;
+  Log::global_log->error()
+      << "Invalid boundary type passed. Check your input file!" << std::endl;
+  return BoundaryType::ERROR;
+}
+
+std::tuple<std::array<double, 3>, std::array<double, 3>>
+BoundaryUtils::getInnerBuffer(const std::array<double, 3> givenRegionBegin,
+                              const std::array<double, 3> givenRegionEnd,
+                              DimensionType dimension, double regionWidth) {
+  std::array<double, 3> returnRegionBegin, returnRegionEnd;
+  for (int i = 0; i < 3; i++) {
+    returnRegionBegin[i] = givenRegionBegin[i];
+    returnRegionEnd[i] = givenRegionEnd[i];
+  }
+
+  const int dimensionLS1 = convertDimensionToLS1Dims(dimension);
+
+  switch (dimension) // can be done with findsign() too, but this is clearer
+  {
+  // in positive case, set the beginning to end-width, or whole domain if width
+  // too large
+  case DimensionType::POSX:
+  case DimensionType::POSY:
+  case DimensionType::POSZ:
+    returnRegionBegin[dimensionLS1] =
+        std::max(returnRegionEnd[dimensionLS1] - regionWidth,
+                 givenRegionBegin[dimensionLS1]);
+    break;
+  // in negative case, set the end to beginning+width, or whole domain if width
+  // too large
+  case DimensionType::NEGX:
+  case DimensionType::NEGY:
+  case DimensionType::NEGZ:
+    returnRegionEnd[dimensionLS1] =
+        std::min(returnRegionBegin[dimensionLS1] + regionWidth,
+                 givenRegionEnd[dimensionLS1]);
+    break;
+
+  default:
+    Log::global_log->error()
+        << "Invalid dimension passed for inner buffer calculation" << std::endl;
+    mardyn_exit(1);
+  }
+  return std::make_tuple(returnRegionBegin, returnRegionEnd);
+}
```

Comment:
```suggestion
std::tuple<std::array<double, 3>, std::array<double, 3>> BoundaryUtils::getInnerBuffer(
  const std::array<double, 3> &givenRegionBegin,
  const std::array<double, 3> &givenRegionEnd,
  DimensionType dimension,
  double regionWidth) {

  std::array<double, 3> returnRegionBegin = givenRegionBegin;
  std::array<double, 3> returnRegionEnd = givenRegionBegin;

  const int dimensionLS1 = convertDimensionToLS1Dims(dimension);
  switch (dimension) {
    // in positive case, set the beginning to end-width, or whole domain if width too large
    case DimensionType::POSX:
      [[fallthrough]];
    case DimensionType::POSY:
      [[fallthrough]];
    case DimensionType::POSZ:
      returnRegionBegin[dimensionLS1] =
        std::max(returnRegionEnd[dimensionLS1] - regionWidth, givenRegionBegin[dimensionLS1]);
      break;
    // in negative case, set the end to beginning+width, or whole domain if width too large
    case DimensionType::NEGX:
      [[fallthrough]];
    case DimensionType::NEGY:
      [[fallthrough]];
    case DimensionType::NEGZ:
      returnRegionEnd[dimensionLS1] =
        std::min(returnRegionBegin[dimensionLS1] + regionWidth, givenRegionEnd[dimensionLS1]);
      break;

    default:
      Log::global_log->error() << "Invalid dimension (" 
                               << dimension 
                               << ") passed for inner buffer calculation\n";
      mardyn_exit(1);
  }
  return {returnRegionBegin, returnRegionEnd};
}
```
1. pass by `&`
2. simpler initialization of return vars
3. `[[fallthrough`
4. formatting
5. Add erroneous value to error message
6. simpler generation of return tuple
7. formatting